lib/traject/macros/met_csv.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module Macros
+  # Macros for extracting values from CSV rows
+  module MetCsv
+    # @param header_or_index [String] the field header or index to accumulate
+
+    DISPLAY_NAME = 'Artist Display Name'
+    SUFFIX = 'Artist Suffix'
+    BEGIN_DATE = 'Artist Begin Date'
+    END_DATE = 'Artist End Date'
+    OBJECT_BEGIN_DATE = 'Object Begin Date'
+    OBJECT_END_DATE = 'Object End Date'
+    ROLE = 'Artist Role'
+    BIO = 'Artist Display Bio'
+    CLASSIFICATION = 'Classification'
+    PUBLIC_DOMAIN = 'Is Public Domain'
+
+    def generate_creator
+      lambda do |row, accumulator, _context|
+        display_pattern = row[DISPLAY_NAME] unless row[DISPLAY_NAME].empty?
+        display_pattern = "#{display_pattern}, #{row[SUFFIX]}" unless row[SUFFIX].empty?
+        display_pattern = "#{display_pattern}, #{row[BEGIN_DATE]}" unless row[BEGIN_DATE].to_s.empty?
```

Comment:
How about extracting `artist_date_range` method?  Additionally, extract `begin_date_present?` and `end_date_present?`

Can you do this? `display_pattern = [display_name, suffix, artist_date_range].select(&:present?).join(', ')`